src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2426,31 +2476,52 @@ private void AddDeclaredNontypeMembers(MembersAndInitializersBuilder builder, Di
-                        AddNonTypeMembers(builder, ((NamespaceDeclarationSyntax)syntax).Members, diagnostics);
+                        AddNonTypeMembers(builder, instanceInitializers: null, ((NamespaceDeclarationSyntax)syntax).Members, diagnostics);
                         break;
 
                     case SyntaxKind.CompilationUnit:
-                        AddNonTypeMembers(builder, ((CompilationUnitSyntax)syntax).Members, diagnostics);
+                        AddNonTypeMembers(builder, instanceInitializers: null, ((CompilationUnitSyntax)syntax).Members, diagnostics);
                         break;
 
                     case SyntaxKind.ClassDeclaration:
                         var classDecl = (ClassDeclarationSyntax)syntax;
-                        AddNonTypeMembers(builder, classDecl.Members, diagnostics);
+                        AddNonTypeMembers(builder, instanceInitializers: noteRecordParameters(classDecl, builder, classDecl.ParameterList), classDecl.Members, diagnostics);
                         break;
 
                     case SyntaxKind.InterfaceDeclaration:
-                        AddNonTypeMembers(builder, ((InterfaceDeclarationSyntax)syntax).Members, diagnostics);
+                        AddNonTypeMembers(builder, instanceInitializers: null, ((InterfaceDeclarationSyntax)syntax).Members, diagnostics);
                         break;
 
                     case SyntaxKind.StructDeclaration:
                         var structDecl = (StructDeclarationSyntax)syntax;
-                        AddNonTypeMembers(builder, structDecl.Members, diagnostics);
+                        AddNonTypeMembers(builder, instanceInitializers: noteRecordParameters(structDecl, builder, structDecl.ParameterList), structDecl.Members, diagnostics);
                         break;
 
                     default:
                         throw ExceptionUtilities.UnexpectedValue(syntax.Kind());
                 }
             }
+
+            static ArrayBuilder<FieldOrPropertyInitializer.Builder>? noteRecordParameters(TypeDeclarationSyntax syntax, MembersAndInitializersBuilder builder, ParameterListSyntax? parameterList)
+            {
+                if (parameterList is null)
+                {
+                    return null;
+                }
+
+                if (builder.RecordConstructorParamList is null)
+                {
+                    builder.RecordConstructorParamList = parameterList;
+                    builder.RecordDeclarationWithParameters = syntax;
+                    builder.InstanceInitializersForRecordDeclarationWithParameters = new ArrayBuilder<FieldOrPropertyInitializer.Builder>();
+                    return builder.InstanceInitializersForRecordDeclarationWithParameters;
+                }
+                else
+                {
+                    // PROTOTYPE: Add error for multiple parameter lists
```

Comment:
> nit: it would be nice to reference the tracking issue for this instead of using PROTOTYPE. [https://github.com/dotnet/roslyn/issues/44676](https://github.com/dotnet/roslyn/issues/44676)

The code was moved from other place in the same file. I didn't add the comment.

---
In reply to: [434735063](https://github.com/dotnet/roslyn/pull/44803#discussion_r434735063) [](ancestors = 434735063)